### PR TITLE
HHH-17375 HQL syntax sugar for array operations

### PR DESCRIPTION
**Pull request by beikov**
https://hibernate.atlassian.net/browse/HHH-17375
https://hibernate.atlassian.net/browse/HHH-18089

### CODE REVIEW

**Comment by gavinking** [comment]
Nice!

File: release-announcement.adoc [hunk 1]
```diff
@@ -98,3 +98,41 @@ class MyEmbeddable { ... }
...
+|`position(element in array)`
+|Overload position function with array_position semantics on array input
+
+|`array_to_string(array, ',', 'null')`
+|`cast(array as String)`
+|Support casting array to string
+
+|`array_contains(array, element)`
+|`array contains element`
+|Contains predicate for containment check
```
Comment (gavinking) [inline release-announcement.adoc:132]: @beikov Note that we also usually allow `cast(.... as String)` to be written as `str(....)`.

File: release-announcement.adoc [hunk 2]
```diff
@@ -98,3 +98,41 @@ class MyEmbeddable { ... }
...
+|`length(array)`
+|Overload length function with array_length semantics on array input
+
+|`array_position(array, element)`
+|`position(element in array)`
+|Overload position function with array_position semantics on array input
+
+|`array_to_string(array, ',', 'null')`
+|`cast(array as String)`
+|Support casting array to string
```
Comment (gavinking) [inline release-announcement.adoc:128]: @beikov Note that we also usually allow `cast(.... as String)` to be written as `str(....)`.

Comment (beikov) [reply release-announcement.adoc:128]: Added a test.

**Pull request merged**